### DISCUSSION TITLE
Use of SingleProfile

### DISCUSSION THREAD

**Discussion by liam-stubbington** [jrkerns/pylinac#410] [Q&A]
Hi, 

I'm really sorry, this is likely quite a basic limit of my python but I am having trouble instantiating a SingleProfile object with a fixed dpmm and no grounding applied. Please see screenshot attached. new_profile is an numpy array with profile values. 

Thanks in advance.
![pylinac_1](https://user-images.githubusercontent.com/71460253/135475685-33aec730-46e9-46a2-bafb-c9f16a9560c2.PNG)

**Comment by jrkerns** [comment]
Hi Liam, 
     The syntax for passing parameters in python can be with or without keywords. So both of these are valid syntax:

`myfunction(myvalue1, myvalue2, myvalue3)`

`myfunction(parameter1=myvalue1, parameter2=myvalue2, parameter3=myvalue3)`

In the case of SingleProfile I see two issues. First is the syntax, but the 2 positional arguments error makes me think you're not on the pylinac version you think you are. 

Pylinac v3 introduced the dpmm and interpolation parameters. In prior versions there was only one parameter you could pass. 

TL;DR:
1. print your pylinac version
`print(pylinac.__version__)`
If it's not 3.0 then upgrade pylinac `pip install pylinac -U` in a command terminal
2. Once on v3, you can call your original expression like so:
`from pylinac.core.profile import SingleProfile, Interpolation`
`myprof = SingleProfile(new_profile, dpmm=2.0, interpolation=Interpolation.LINEAR, ground=False)`

**Reply by liam-stubbington** [reply]
Yep - pylinac version not 3.0. 
Thanks James.
